src/pages/profile/index.jsx

```diff
@@ -0,0 +1,94 @@
+import Card from "../../components/card";
+import UserDetails from "../../components/UserDetails";
+import UserProfileIcon from "../../components/UserProfileIcon";
+import TextInput from "../../components/form/textInput";
+import Form from "../../components/form";
+import "./profile.css";
+import useUser from "../../hooks/useUser";
+import { useState, useEffect } from "react";
+
+const Profile = () => {
+  const [formData, setFormData] = useState({
+    firstName: "",
+    lastName: "",
+    username: "",
+    githubUsername: "",
+    role: "",
+    cohortId: "",
+    email: "",
+    bio: "",
+  });
+
+  const { currentUser } = useUser();
+
+  useEffect(() => {
+    setFormData(currentUser);
+  }, [currentUser]);
+
+  const handleSubmit = (e) => {};
+
+  const handleChange = (e) => {};
+
+  const labelMaker = (string) => {
+    let result;
+    switch (string) {
+      case "cohortId":
+        result = "Cohort ID";
+        break;
+      case "role":
+        result = "Role";
+        break;
+      case "email":
+        result = "Email";
+        break;
+      case "firstName":
+        result = "First Name";
+        break;
+      case "lastName":
+        result = "Last Name";
+        break;
+      case "bio":
+        result = "Bio";
+        break;
+      case "githubUsername":
+        result = "GitHub Username";
+        break;
+      default:
+        result = "Something went wrong";
+    }
+    return result;
+  };
```

Comment:
I think an object map would be a little cleaner here:

```js
const labelMap = {
  cohortId: 'Cohort ID',
  role: 'Role',
  ...
}

...

label={labelMap[input]}
```

Reply:
Great! Updated.